src/controlplane/scraper.go

```diff
@@ -100,24 +106,31 @@ func (s *Scraper) Run(i *integration.Integration) error {
 		for _, autodiscover := range component.AutodiscoverConfigs {
 			podName, err := s.discoverPod(autodiscover)
 			if err != nil {
-				return fmt.Errorf("control plane component %s discovery failed: %v", component.Name, err)
+				return fmt.Errorf("control plane component %q discovery failed: %v", component.Name, err)
 			}
 
 			if podName == "" {
-				s.logger.Debugf("No pod found for component: %s", component.Name)
+				s.logger.Debugf("No %q pod found with labels %q ", component.Name, autodiscover.Selector)
 				continue
 			}
 
-			// TODO the fallback mechanism to a different endpoint for the same pod is not implemented.
+			connector, err := controlplaneClient.DefaultConnector(
+				autodiscover.Endpoints,
+				s.K8s,
+				s.inClusterConfig,
+				s.logger,
+			)
+			if err != nil {
+				return fmt.Errorf("control plane component %q failed creating connector: %v", component.Name, err)
+			}
+
 			client, err := controlplaneClient.New(controlplaneClient.Config{
```

Comment:
Each iteration we need to create the clients, I guess this is needed since the endpoints could change. 

In KSM the client is not created each time and it is a simple http pester, I guess we cannot do the same due to auth.

I realize that CP as both complexity of KSM (where to connect) and Kubelet (how to connect)

With informers this task is not too demanding, however, I would add some comments to explain why the approach is a bit different

Reply:
I have divided this into autodicover and external endpoint functions with descriptions , let me know if it's clear enough otherwise i could add some inline comments.